dom/src/main/java/org/apache/james/mime4j/message/MessageBuilder.java

```diff
@@ -75,15 +75,68 @@
     private BodyFactory bodyFactory;
     private boolean flatMode;
     private boolean rawContent;
+    private final boolean ensureDateNotNull;
 
+    /**
+     * Deprecated: please use {@link #of() of} instead
+     * Building a builder using this method will result in a buggy builder which
+     * will create message that do not respect the <code>Message.getDate()</code> contract regarding
+     * the return value when the message do not have a Date header
+     * See <a href="https://issues.apache.org/jira/browse/MIME4J-262">MIME4J-262</a>
+     */
+    @Deprecated
+    public MessageBuilder() {
+        this(true);
+    }
+
+    private MessageBuilder(boolean ensureDateNotNull) {
+        this.ensureDateNotNull = ensureDateNotNull;
+    }
+
+    public static MessageBuilder of() {
+        return new MessageBuilder(false);
+    }
+
+    /**
+     * Deprecated: please use {@link #of() of} instead
+     * Building a builder using this method will result in a buggy builder which
+     * will create message that do not respect the <code>Message.getDate()</code> contract regarding
+     * the return value when the message do not have a Date header
+     * See <a href="https://issues.apache.org/jira/browse/MIME4J-262">MIME4J-262</a>
+     */
+    @Deprecated
     public static MessageBuilder create() {
-        return new MessageBuilder();
+        return new MessageBuilder(true);
```

Comment:
What true / false means in these builders parameter? (extract true and false to constants to name them as it is confusing)